app/models/case_log.rb

```diff
@@ -536,7 +536,9 @@ def reset_derived_questions
     dependent_questions = { waityear: [{ key: :renewal, value: 0 }],
```

Comment:
Do you remember why we needed to hardcode these dependencies here rather than being able to derive them from the question definition?

Reply:
not exactly, the first thing that comes to mind is that maybe validations were failing for some reason if we didn't reset these. Do you?

Reply:
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/322 bad PR descriptions come back to haunt me

Reply:
Looks like it's because it's a slightly special case. Usually in the definition when we says a question `depends_on` something, when it's not met it's not routed to *and* any answer is cleared.

In these cases, it *is* routed to but we still want an inferred answer cleared.

Source. https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/322


This probably warrants becoming it's own more explicit thing but for the time being maybe a comment in the code here is better than nothing?

Reply:
`So we're already getting the depends on to determine if something is not routed to and if it's not we clear it.
In this case we want to clear something that is routed to, but only if it's one of the fields we're inferring. We can get the questions that are routed to from this JSON, but we would still need to filter them down to the ones that get inferred, no?`
good thing we got comments